### ISSUE TITLE
relateTo()  removes existing relations

### ISSUE THREAD

**Issue by ptrvif** [adam-cowley/neode#91]
Hello,

I have my model defined as:

```
module.exports = {
  id: {
    type: 'uuid',
    primary: true
  },
  name:  {
    type: 'string',
    index: true
  },

  sends_to: {
    type: 'relationships',
    target: 'Destination',
    relationship: 'SENDS_TO',
    direction: 'out',
    properties: {},
    eager: true
  },

  receives_from: {
    type: 'relationships',
    target: 'Destination',
    relationship: 'RECEIVES_FROM',
    direction: 'in',
    properties: {},
    eager: true
  }

}
```

When I create a new node and call toJson(), I can see relationships properties with empty arrays for relationships properties like:

```
{
"_id":35,
"_labels":["Server"],
"id":"c2192eb8-fee8-4a71-b274-361194299254",
"name":"testserver",
"sends_to":[],
"receives_from":[]
}
```

But as soon as I call 'relateTo()' method, the relationship property is removed.
I suppose it is because of the following code in   Node.js->relateTo()  method:

```
return RelateTo(this._neode, this, node, relationship, properties, force_create)
            .then(rel => {
                **this._eager.delete(type);**

                return rel;
            })
```

Is it done in purpose ? What is the reason behind it?
Would not it be right just to add a new relation to the collection instead of clearing the entire property (all existing relations)?

**Comment by adam-cowley** [comment]
Yeah fair point - the trade off I made at the time was it's better to not have any relationships than they be out of date.  This should definitely be improved.  Fancy submitting a PR?

**Comment by ptrvif** [comment]
Adam,
thank you for your response. 
I just wanted to confirm my understanding to make sure I did not miss something. I do totally understand the point about trade off , I would do the same.
I will consider working on it , this time of the year fairly busy , but I really like Neo4j and this library is pretty cool , so I will try allocate some time to it.
Thanks again.

BTW, I'd like your opinion on some other idea ..if you don't mind. My use case is probably not very typical, but what I have is a large number of input requests which result in much lower number of Graph DB updates. So in my REST service I added a layer of in-memory cache for Nodes and Relations.  
Would it be a useful feature if it was supported by this library 'out-of-box' ?   Or this is rather something not typically needed..

Thanks!
